server/src/main/java/org/apache/druid/client/CachingQueryRunner.java

```diff
@@ -77,20 +83,15 @@ public CachingQueryRunner(
   {
     Query<T> query = queryPlus.getQuery();
     final CacheStrategy strategy = toolChest.getCacheStrategy(query);
-    final boolean populateCache = CacheUtil.isPopulateSegmentCache(
-        query,
-        strategy,
-        cacheConfig,
-        CacheUtil.ServerType.DATA
-    );
-    final boolean useCache = CacheUtil.isUseSegmentCache(query, strategy, cacheConfig, CacheUtil.ServerType.DATA);
+    final boolean populateCache = canPopulateCache(query, strategy);
+    final boolean useCache = canUseCache(query, strategy);
 
     final Cache.NamedKey key;
-    if (strategy != null && (useCache || populateCache)) {
```

Comment:
Is it not possible for `strategy` to be null?

Reply:
if its null, then both `useCache` and `populateCache` will be set to false. if that doesn't hold, then an NPE can occur later in the code. I can still add another check in `canUseCache`, `canPopulateCache`.

Reply:
Got it. I don't think the extra check is necessary, given what you just said. Maybe just a comment explaining that `strategy` can't be null if either of the cache flags is true.